### PR TITLE
Move vagrant-ci from CentOS 7 to CentOS 9 Stream

### PR DESCRIPTION
**Pull request by vpodzime**
CentOS 9 Stream has new enough PCRE2 required by the no-deps build. And it's generally a more future-proof option than CentOS 7.

### CODE REVIEW

File: contrib/vagrant-ci/centos-9s-x64/Vagrantfile [hunk 1]
```diff
@@ -73,10 +75,11 @@ SHELL
 
     fd.vm.provision "shell", inline: <<-SHELL
 set -e
-yum -q -y install http://opensource.wandisco.com/centos/7/git/x86_64/wandisco-git-release-7-1.noarch.rpm || true
-yum -q -y install 'perl(ExtUtils::MakeMaker)' 'perl(Digest::MD5)' 'perl(Module::Load::Conditional)' 'perl(IO::Uncompress::Gunzip)' 'perl(JSON::PP)' 'perl(IPC::Cmd)' ncurses-devel rpm-build
-yum -q -y install psmisc # for fuser command
-yum -q -y install wget
+dnf -q -y install http://opensource.wandisco.com/centos/7/git/x86_64/wandisco-git-release-7-1.noarch.rpm || true
```
Comment (nickanderson) [inline contrib/vagrant-ci/centos-9s-x64/Vagrantfile:78]: Just curious, why do we use wandisco-git, do you know?

**Review by nickanderson — APPROVED** [review]
Makes sense to me. Just a question about why wandisco-git.

File: contrib/vagrant-ci/centos-9s-x64/Vagrantfile (hunk 1 above)
Comment (vpodzime) [reply contrib/vagrant-ci/centos-9s-x64/Vagrantfile:78]: Good catch! This was needed on CentOS 7 to get a newer git there. Should not be needed anymore.

**Review by craigcomstock — APPROVED** [review]
Cool. Will be interesting to see how this differs from the rhel-9 we use in jenkins. If at all.

File: contrib/vagrant-ci/centos-9s-x64/Vagrantfile (hunk 1 above)
Comment (nickanderson) [reply contrib/vagrant-ci/centos-9s-x64/Vagrantfile]: Sweet, a useful review.

**Review by nickanderson — APPROVED** [review]

**Pull request merged**